pkg/provisioner/controller.go

```diff
@@ -606,25 +606,25 @@ func (c *obcController) deleteResources(ob *v1alpha1.ObjectBucket, cm *corev1.Co
 }
 
 // Add finalizer and labels to the OBC.
-func (c *obcController) setOBCMetaFields(obc *v1alpha1.ObjectBucketClaim) (err error) {
+func (c *obcController) setOBCMetaFields(obc *v1alpha1.ObjectBucketClaim) (*v1alpha1.ObjectBucketClaim, error) {
 	clib := c.libClientset
 
 	logD.Info("getting OBC to set metadata fields")
-	obc, err = clib.ObjectbucketV1alpha1().ObjectBucketClaims(obc.Namespace).Get(context.TODO(), obc.Name, metav1.GetOptions{})
+	obcFetched, err := clib.ObjectbucketV1alpha1().ObjectBucketClaims(obc.Namespace).Get(context.TODO(), obc.Name, metav1.GetOptions{})
```

Comment:
1) why does the OBC need to be retrieved again here? Nothing has changed has it (other than perhaps a window where the user changes it right after creating it)?
2) Why can't we append the lib label to the existing label(s) in the OBC?

Reply:
I had a similar question. Existing code was retrieving the OBC again. Do we need to do this in case the OBC gets updated twice in rapid succession by the user before this reconcile happens?

Reply:
@BlaineEXE To answer your question about an OBC that's created -> lib GETs it -> user edits it -> lib's copy is stale:  do we still ignore update events on the OBC? If yes then this window exists and it seems that we would lose the user's changes. But, even with the GET here we still have the same window, just smaller, right?

Reply:
Yes, I think that's right. If the user updates twice rapidly, then a second reconcile will still be queued to happen again after this reconcile finishes IIUC. So this isn't necessary. I think @jeffvance and I are agreed that this `Get`, despite being in the repo pre-existing, should be removed. 

@llamerada-jp you can look at some of the other functions to see how update attempts use `DeepCopy`-ed objects to not modify the object used as input `obc` in this case.

Reply:
I update it using DeepCopy.